File: algorithms/1_easy/R/01844_replace_all_digits_with_characters/solution.py
class Solution:
    def replaceDigits(self, s: str) -> str:
        n = len(s)
        li = []

        for i in range(1, n, 2):
            li.append(s[i - 1])
            li.append(chr(ord(s[i]) - ord('0') + ord(s[i - 1])))

        if n % 2 == 1:
            li.append(s[-1])

        return "".join(li)

File: algorithms/1_easy/R/01844_replace_all_digits_with_characters/solution.cpp
class Solution {
public:
    string replaceDigits(string s) {
        const size_t len{s.size()};

        for (size_t i{1}; i < len; i += 2) {
            s[i] = s[i] - '0' + s[i - 1];
        }

        return s;
    }
};
